File: 1694-make-sum-divisible-by-p/make-sum-divisible-by-p.py
class Solution:
    def minSubarray(self, nums: List[int], p: int) -> int:
        lookup = {0: -1}
        total_sum = sum(nums)
        total_rem = total_sum % p
        if (total_rem == 0):
            return 0

        output = len(nums)
        current_sum = 0

        for i in range(len(nums)):
            current_sum += nums[i]
            current_rem = current_sum % p

            prefix_rem = ((current_rem - total_rem) + p) % p

            if (prefix_rem in lookup):
                output = min(output, i - lookup[prefix_rem])
            
            lookup[current_rem] = i

        return - 1 if output == len(nums) else output

File: 1694-make-sum-divisible-by-p/make-sum-divisible-by-p.js
var minSubarray = function(nums, p) {
    let total = nums.reduce((a, b)=> a + b, 0)
    if (total % p == 0) return 0
    let totalRemainder = total % p
    let lookup = {0: -1}
    let minLength = nums.length
    let currSum = 0

    for (let i = 0; i < nums.length; i++) {
        currSum += nums[i]
        let currRemainder = currSum % p

        targetRemainder = ((currRemainder - totalRemainder) + p) % p
        if (targetRemainder in lookup) {
            minLength = Math.min(minLength, i - lookup[targetRemainder])
        }
        lookup[currRemainder] = i

    }
    return minLength == nums.length ? -1 : minLength

};